decidim-core/app/commands/decidim/create_omniauth_registration.rb

```diff
@@ -19,45 +20,59 @@ def call
       verify_oauth_signature!
 
       begin
-        if identity.present?
-          @user = identity.user
-        else
-          return broadcast(:invalid) if form.invalid?
+        return broadcast(:invalid) if form.invalid?
 
-          create_user
-          create_identity
+        transaction do
+          create_or_find_identity
+          create_or_find_user
         end
 
         broadcast(:ok, @user)
-      rescue ActiveRecord::RecordInvalid, ActiveRecord::RecordNotUnique
+      rescue ActiveRecord::RecordInvalid
+        broadcast(:invalid)
+      rescue ActiveRecord::RecordNotUnique
         broadcast(:error)
       end
     end
 
     private
 
-    attr_reader :form
+    attr_reader :form, :verified_email
 
-    def create_user
+    def create_or_find_user
       generated_password = SecureRandom.hex
+      @user = identity.try(:user) || User.find_or_initialize_by(
+        email: verified_email,
+        organization: organization
+      )
 
-      @user = User.create!(email: form.email,
-                           name: form.name,
-                           password: generated_password,
-                           password_confirmation: generated_password,
-                           organization: form.current_organization,
-                           tos_agreement: "1")
+      unless @user.persisted?
+        @user.email = (verified_email || form.email)
+        @user.name = form.name
+        @user.password = generated_password
+        @user.password_confirmation = generated_password
+      end
 
-      @user.skip_confirmation! if form.email_verified?
+      @user.skip_confirmation!
+      @user.tos_agreement = "1"
+      @user.save!
     end
 
     def create_identity
```

Comment:
Rename this to `create_or_find_identity` (specs are complaining about this, since you are calling this method but does not exist)